src/hooks/useFirestore.js

```diff
@@ -5,6 +5,9 @@ const useFirestore = (collection) => {
   const [docs, setDocs] = useState([]);
 
   useEffect(() => {
+    if (docs.length > 0) {
```

Comment:
this is a solution that I found on Stack Overflow but it doesn't seem to be solving our unmounting warning. I found another solution that I think might actually solve it
```suggestion
    let isMounted = true;

    const unsubscribe = db.collection(collection).onSnapshot((snapshot) => {
      let documents = [];
      snapshot.forEach((doc) => {
        documents.push({ ...doc.data(), id: doc.id });
      });
      if(isMounted) setDocs(documents);
      return () => {
        unsubscribe();
      };
    });

    return () => { isMounted = false };
```